yb-voyager/src/query/queryissue/detectors.go

```diff
@@ -195,3 +199,46 @@ func (d *RangeTableFuncDetector) GetIssues() []QueryIssue {
 	}
 	return issues
 }
+
+type SelectStmtDetector struct {
+	query                       string
+	limitOptionWithTiesDetected bool
+}
+
+func NewSelectStmtDetector(query string) *SelectStmtDetector {
+	return &SelectStmtDetector{
+		query: query,
+	}
+}
+
+func (d *SelectStmtDetector) getSelectStmtFromProto(msg protoreflect.Message) (*pg_query.SelectStmt, error) {
+	protoMsg := msg.Interface().(proto.Message)
+	selectStmtNode, ok := protoMsg.(*pg_query.SelectStmt)
+	if !ok {
+		return nil, fmt.Errorf("failed to cast %s to %s", queryparser.PG_QUERY_SELECTSTMT_NODE, queryparser.PG_QUERY_SELECTSTMT_NODE)
+	}
+	return selectStmtNode, nil
+}
+
+func (d *SelectStmtDetector) Detect(msg protoreflect.Message) error {
+	if queryparser.GetMsgFullName(msg) == queryparser.PG_QUERY_SELECTSTMT_NODE {
+		selectStmtNode, err := d.getSelectStmtFromProto(msg)
+		if err != nil {
+			return err
+		}
+		// checks if a SelectStmt node uses a FETCH clause with TIES
+		// https://www.postgresql.org/docs/13/sql-select.html#SQL-LIMIT
+		if selectStmtNode.LimitOption == pg_query.LimitOption_LIMIT_OPTION_WITH_TIES {
+			d.limitOptionWithTiesDetected = true
+		}
+	}
+	return nil
+}
```

Comment:
I am thinking can we stick to one of the approaches (protomsg) as a primary approach, and use other one where its significantly making things easy, other code might become a bit messy. 
There is a overhead of conversion also comes.


Here in this case i guess something like `GetStringField()` or `GetEnumField()` we already have, can be used to get pg_query.LimitOption_LIMIT_OPTION_WITH_TIES

Reply:
I see your point about consistency. But why, in general, do you prefer accessing fields of a message via protoreflect.Message rather than structs directly? 

>There is a overhead of conversion also comes.
I doubt there is significant overhead. It's just type conversion.

Reply:
Merged this for now, since it's been pending for a while and had to resolve a lot of conflicts! 
But let's discuss this nonetheless.

Reply:
> But why, in general, do you prefer accessing fields of a message via protoreflect.Message rather than structs directly?

@makalaaneesh 
Couple of reasons:
The field name we use to fetch from the protomsg is exactly the same as visible in a parse tree output.
In case of structs, you need to figureout the struct field which gives you the required info(sligtly different named)

Consistency here is another reason.